docs/website/docs/extensions/index.md

```diff
@@ -0,0 +1,442 @@
+# Extension mechanisms
+
+**NOTE**: much of this describes provisions for extension within IREE but until
+the core of the system has settled little work will be done to fully flesh-out
+and document them in detail. A large majority of things that would make someone
+want to extend IREE can instead be accomplished much easier and performantly
+using native MLIR dialects that are then processed by the IREE compiler.
```

Comment:
if you want, you can use an admonition (https://squidfunk.github.io/mkdocs-material/reference/admonitions/) for this

![image](https://user-images.githubusercontent.com/4010439/139780005-e6f0bade-a0bc-45af-95f7-c6142e67114e.png)


```suggestion
!!! Note
    Much of this describes provisions for extension within IREE but until
    the core of the system has settled little work will be done to fully
    flesh-out and document them in detail. A large majority of things that would
    make someone want to extend IREE can instead be accomplished much easier and
    performantly using native MLIR dialects that are then processed by the IREE
    compiler.
```